### PR TITLE
Add new autodiscovery docs section for k8s

### PR DESCRIPTION
**Pull request by ChrsMark**
Closes https://github.com/elastic/observability-dev/issues/2386
Fixes https://github.com/elastic/observability-docs/issues/2195

### CODE REVIEW

File: docs/en/ingest-management/elastic-agent/configuration/elastic-agent-kubernetes-autodiscovery.asciidoc [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+but here we provide some directions to speed up your investigation:
+
+1. Exec inside an Agent's Pod and run inspect command to verify how inputs are constructed dynamically:
+
+["source", "sh", subs="attributes"]
+------------------------------------------------
+./elastic-agent inspect -c /etc/elastic-agent/agent.yml output -o default -v -d "*"
+------------------------------------------------
+
+Specifically check how the inputs are being populated.
```
Comment (dedemorton) [inline docs/en/ingest-management/elastic-agent/configuration/elastic-agent-kubernetes-autodiscovery.asciidoc:302]:
```suggestion
. Exec inside an Agent's Pod and run inspect command to verify how inputs are constructed dynamically:
+
["source", "sh", subs="attributes"]
------------------------------------------------
./elastic-agent inspect -c /etc/elastic-agent/agent.yml output -o default -v -d "*"
------------------------------------------------
+
Specifically check how the inputs are being populated.
```

File: docs/en/ingest-management/elastic-agent/configuration/elastic-agent-kubernetes-autodiscovery.asciidoc [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+
+["source", "sh", subs="attributes"]
+------------------------------------------------
+tail -f /etc/elastic-agent/data/logs/elastic-agent-*.ndjson
+------------------------------------------------
+
+Verify that the hints feature is enabled in the config and look for hints' related logs like:
+"Generated hints mappings are ..."
+In this, you can find what are the mappings that are extracted out of the annotations and have an indicator
+if these values are able to populate a specific input.
```
Comment (dedemorton) [inline docs/en/ingest-management/elastic-agent/configuration/elastic-agent-kubernetes-autodiscovery.asciidoc:314]:
```suggestion
. Check the logs of Elastic Agent:
+
["source", "sh", subs="attributes"]
------------------------------------------------
tail -f /etc/elastic-agent/data/logs/elastic-agent-*.ndjson
------------------------------------------------
+
Verify that the hints feature is enabled in the config and look for hints' related logs like:
"Generated hints mappings are ..."
In this, you can find what are the mappings that are extracted out of the annotations and have an indicator
if these values are able to populate a specific input.
```

File: docs/en/ingest-management/elastic-agent/configuration/elastic-agent-kubernetes-autodiscovery.asciidoc [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+
+3. Check the logs of Metricbeat/Filebeat:
+
+["source", "sh", subs="attributes"]
+------------------------------------------------
+tail -f /etc/elastic-agent/data/logs/default/metricbeat-*.ndjson
+------------------------------------------------
```
Comment (dedemorton) [inline docs/en/ingest-management/elastic-agent/configuration/elastic-agent-kubernetes-autodiscovery.asciidoc:321]:
```suggestion
. Check the logs of Metricbeat/Filebeat:
+
["source", "sh", subs="attributes"]
------------------------------------------------
tail -f /etc/elastic-agent/data/logs/default/metricbeat-*.ndjson
------------------------------------------------
```

File: docs/en/ingest-management/elastic-agent/configuration/elastic-agent-kubernetes-autodiscovery.asciidoc [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+
+4. Check how the target input template looks like, ie:
+
+["source", "sh", subs="attributes"]
+------------------------------------------------
+cat -f /etc/elastic-agent/inputs.d/redis.yaml
+------------------------------------------------
```
Comment (dedemorton) [inline docs/en/ingest-management/elastic-agent/configuration/elastic-agent-kubernetes-autodiscovery.asciidoc:328]:
```suggestion
. Check how the target input template looks like, ie:
+
["source", "sh", subs="attributes"]
------------------------------------------------
cat -f /etc/elastic-agent/inputs.d/redis.yaml
------------------------------------------------
```

**Review by dedemorton — COMMENTED** [review]
Applying these suggestions should fix your doc build errors.

**Comment by dedemorton** [comment]
I'll do a proper review early next week after these docs are building.

**Comment by dedemorton** [comment]
@ChrsMark It looks like the original content was never edited by a writer. Rather than carpet bombing your review with edits, it might be better for me to focus on the content that has actually changed in this PR and then follow up with edits in a separate PR. Can you tell me which lines have actually changed (to save me time doing a side-by-side compare), or let me know if you just want me to do the edits as suggestions. Thanks!

**Comment by ChrsMark** [comment]
@dedemorton feel free to provide suggestions in the content of this PR. As you can see there were 2 places that we were mentioning about "autodiscovery", and with this PR the goal is to move everything related to "autodiscovery" into a single place.

File: docs/en/ingest-management/elastic-agent/configuration/elastic-agent-kubernetes-autodiscovery.asciidoc [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+
+After deploying this Pod the data will start flowing in automatically. Note that at this point all the assets
+(dashboards, ingest pipelines etc) related to the Redis integration are not installed and one would need to
+explicitly install them through Kibana.
+
+[discrete]
+=== Hints autodiscovery: Troubleshooting (standalone agent)
+
+When things do not work as expected one would need to troubleshoot their setup. In that case many things can be wrong
+but here we provide some directions to speed up your investigation:
```
Comment (dedemorton) [inline docs/en/ingest-management/elastic-agent/configuration/elastic-agent-kubernetes-autodiscovery.asciidoc:293]:
```suggestion
When things do not work as expected, you may need to troubleshoot your setup. Here we provide some directions to speed up your investigation:
```

File: docs/en/ingest-management/elastic-agent/configuration/elastic-agent-kubernetes-autodiscovery.asciidoc [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+
+. Exec inside an Agent's Pod and run inspect command to verify how inputs are constructed dynamically:
```
Comment (dedemorton) [inline docs/en/ingest-management/elastic-agent/configuration/elastic-agent-kubernetes-autodiscovery.asciidoc:295]:
```suggestion
. Exec inside an Agent's Pod and run the `inspect` command to verify how inputs are constructed dynamically:
```

File: docs/en/ingest-management/elastic-agent/configuration/elastic-agent-kubernetes-autodiscovery.asciidoc [hunk 7]
```diff
@@ -0,0 +1,328 @@
...
++
+["source", "sh", subs="attributes"]
+------------------------------------------------
+./elastic-agent inspect -c /etc/elastic-agent/agent.yml output -o default -v -d "*"
+------------------------------------------------
++
+Specifically check how the inputs are being populated.
+
+. Check the logs of Elastic Agent:
```
Comment (dedemorton) [inline docs/en/ingest-management/elastic-agent/configuration/elastic-agent-kubernetes-autodiscovery.asciidoc:304]:
```suggestion
. View the {agent} logs:
```

File: docs/en/ingest-management/elastic-agent/configuration/elastic-agent-kubernetes-autodiscovery.asciidoc [hunk 8]
```diff
@@ -0,0 +1,328 @@
...
++
+["source", "sh", subs="attributes"]
+------------------------------------------------
+tail -f /etc/elastic-agent/data/logs/elastic-agent-*.ndjson
+------------------------------------------------
++
+Verify that the hints feature is enabled in the config and look for hints' related logs like:
+"Generated hints mappings are ..."
+In this, you can find what are the mappings that are extracted out of the annotations and have an indicator
+if these values are able to populate a specific input.
```
Comment (dedemorton) [inline docs/en/ingest-management/elastic-agent/configuration/elastic-agent-kubernetes-autodiscovery.asciidoc:314]: Here's my attempt at an edit, but without seeing the log entries, the details here are a bit hard to parse.

```suggestion
Verify that the hints feature is enabled in the config and look for hints-related logs like:
"Generated hints mappings are ..."
In these logs, you can find the mappings that are extracted out of the annotations and determine
if the values are able to populate a specific input.
```

File: docs/en/ingest-management/elastic-agent/configuration/elastic-agent-kubernetes-autodiscovery.asciidoc [hunk 9]
```diff
@@ -0,0 +1,328 @@
...
+
+. Check the logs of Metricbeat/Filebeat:
```
Comment (dedemorton) [inline docs/en/ingest-management/elastic-agent/configuration/elastic-agent-kubernetes-autodiscovery.asciidoc:316]:
```suggestion
. View the {metricbeat} and {filebeat} logs:
```

File: docs/en/ingest-management/elastic-agent/configuration/elastic-agent-kubernetes-autodiscovery.asciidoc [hunk 10]
```diff
@@ -0,0 +1,328 @@
...
++
+["source", "sh", subs="attributes"]
+------------------------------------------------
+tail -f /etc/elastic-agent/data/logs/default/metricbeat-*.ndjson
+------------------------------------------------
+
+. Check how the target input template looks like, ie:
```
Comment (dedemorton) [inline docs/en/ingest-management/elastic-agent/configuration/elastic-agent-kubernetes-autodiscovery.asciidoc:323]:
```suggestion
. View the target input template, for example:
```

File: docs/en/ingest-management/elastic-agent/configuration/providers/kubernetes-provider.asciidoc [hunk 11]
```diff
@@ -510,4 +326,8 @@ The available keys are:
 |`kubernetes.selectors.*`
 |`string`
 |Kubernetes selectors
-|===
\ No newline at end of file
+|===
+
+To learn more about how you can leverage these variables for autodiscovery refer to
+<<elastic-agent-kubernetes-autodiscovery,kubernetes autodiscovery with Elastic Agent>> for more information
+about shaping dynamic inputs for autodiscovery.
```
Comment (dedemorton) [inline docs/en/ingest-management/elastic-agent/configuration/providers/kubernetes-provider.asciidoc:333]:
```suggestion
To learn more about how you can leverage these variables for autodiscovery, refer to
<<elastic-agent-kubernetes-autodiscovery,kubernetes autodiscovery with Elastic Agent>> for more information
about shaping dynamic inputs for autodiscovery.
```

Comment (dedemorton) [reply docs/en/ingest-management/elastic-agent/configuration/providers/kubernetes-provider.asciidoc:333]: I feel like the last part of this sentence tacks on a new idea and makes the sentence harder to parse. Can you drop either the introductory phrase ("To learn more about how you can leverage these variables for autodiscovery") or the end ("for more information about shaping dynamic inputs for autodiscovery") without losing meaning here?

File: docs/en/ingest-management/elastic-agent/configuration/elastic-agent-kubernetes-autodiscovery.asciidoc [hunk 12]
```diff
@@ -0,0 +1,328 @@
...
+but here we provide some directions to speed up your investigation:
+
+. Exec inside an Agent's Pod and run inspect command to verify how inputs are constructed dynamically:
++
+["source", "sh", subs="attributes"]
+------------------------------------------------
+./elastic-agent inspect -c /etc/elastic-agent/agent.yml output -o default -v -d "*"
+------------------------------------------------
++
+Specifically check how the inputs are being populated.
```
Comment (dedemorton) [inline docs/en/ingest-management/elastic-agent/configuration/elastic-agent-kubernetes-autodiscovery.asciidoc:302]:
```suggestion
Specifically examine how the inputs are being populated.
```

Comment (dedemorton) [reply docs/en/ingest-management/elastic-agent/configuration/elastic-agent-kubernetes-autodiscovery.asciidoc:302]: It's kind of a picky editor comment, but it's best to avoid the word "check" because it has multiple meanings.

**Review by dedemorton — COMMENTED** [review]
Thanks for updating the docs! Most of these are editorial changes for style, consistency, clarity, and conciseness.

There are a lot of suggestions here, so I suggest that you add them to a batch and commit them in a single commit.

File: docs/en/ingest-management/elastic-agent/configuration/elastic-agent-kubernetes-autodiscovery.asciidoc [hunk 13]
```diff
@@ -0,0 +1,328 @@
...
+
+[float]
+===== `co.elastic.hints/timeout`
+
+Metrics retrieval timeout, ie: 3s
+
+[float]
+===== `co.elastic.hints/username`
+
+The username to use for authentication
```
[the remark this answers is not in the document]
Comment (ChrsMark) [reply docs/en/ingest-management/elastic-agent/configuration/elastic-agent-kubernetes-autodiscovery.asciidoc:183]: Not yet :), in general this list will be expanded with the time being.

File: docs/en/ingest-management/elastic-agent/configuration/elastic-agent-kubernetes-autodiscovery.asciidoc [hunk 14]
```diff
@@ -0,0 +1,328 @@
...
+[float]
+===== `co.elastic.hints/stream`
+
+The stream to use for logs collection, ie stdout/stderr.
+
+In any case if the specified package has no logs support a generic container's logs input will be used as a
+fallback.
+
+[discrete]
+=== Available packages that support hints autodiscovery (standalone agent)
```
[the remark this answers is not in the document]
Comment (ChrsMark) [reply docs/en/ingest-management/elastic-agent/configuration/elastic-agent-kubernetes-autodiscovery.asciidoc]: Fair point. I will remove this extra note.

**Comment by ChrsMark** [comment]
Thanks for the review @dedemorton ! Comments/suggestions have been addressed, feel free to have another look when you have the time :).

File: docs/en/ingest-management/elastic-agent/configuration/elastic-agent-kubernetes-autodiscovery.asciidoc [hunk 15]
```diff
@@ -0,0 +1,326 @@
...
+for the container. Hints tell {agent} how to monitor the container by using the proper integration.
+This is the full list of supported hints:
+
+[float]
+===== `co.elastic.hints/package`
+
+The package to use for monitoring.
+
+[float]
+===== `co.elastic.hints/data_stream`
```
Comment (ChrsMark) [inline docs/en/ingest-management/elastic-agent/configuration/elastic-agent-kubernetes-autodiscovery.asciidoc:152]: This should be optional.

**Review by dedemorton — APPROVED** [review]
LGTM (Looks Groovy to Me!)

Thanks again for doing this. I like how the docs are more prominent in the navigation now.

File: docs/en/ingest-management/elastic-agent/configuration/elastic-agent-kubernetes-autodiscovery.asciidoc [hunk 16]
```diff
@@ -0,0 +1,326 @@
+[[elastic-agent-kubernetes-autodiscovery]]
+= Kubernetes autodiscovery with {agent}
+
+[discrete]
+== Autodiscover targeted Pods based on conditions
```
Comment (ChrsMark) [inline docs/en/ingest-management/elastic-agent/configuration/elastic-agent-kubernetes-autodiscovery.asciidoc:5]: `Conditions based autodiscover`.

File: docs/en/ingest-management/elastic-agent/configuration/elastic-agent-kubernetes-autodiscovery.asciidoc [hunk 17]
```diff
@@ -0,0 +1,326 @@
...
+    - add_fields:
+        fields:
+          cluster:
+            name: kind-kind
+            url: https://127.0.0.1:52500
+        target: orchestrator
+----
+
+[discrete]
+== Autodiscover targeted Pods using hints
```
Comment (ChrsMark) [inline docs/en/ingest-management/elastic-agent/configuration/elastic-agent-kubernetes-autodiscovery.asciidoc:138]: `Hints (annotations) based autodiscover`
